metrics.go

```diff
@@ -16,31 +17,65 @@ import (
 const maxLabelValueLength = 80
 
 var (
-	filepath string
+	filepath          string
+	isValidHostHeader = regexp.MustCompile(`^[a-z0-9.-]+$`).MatchString
 )
 
-func sanitizeValue(label string, value interface{}) string {
+func sanitizeLabel(label string, value interface{}) (string, string) {
 
-	// Convert to a string, no matter what underlying type it is
-	var labelValue string
-	if value != nil {
-		labelValue = fmt.Sprintf("%v", value)
+	if value == nil || value == "" || value == "-" {
+		return label, ""
 	}
 
+	// Convert to a string, no matter what underlying type it is
+	labelValue := fmt.Sprintf("%v", value)
+	labelValue = strings.TrimSpace(labelValue)
+
 	switch label {
 	case "content_type":
-		labelValue = strings.Split(labelValue, ";")[0]
+		label = "content_type_bucket"
```

Comment:
In hingsight, I don't think I like this as much as keeping `labelValue` aligned with the actual content-type.